### DISCUSSION TITLE
why ssh_autodetect.py sometimes wrong sometimes fine!!!

### DISCUSSION THREAD

**Discussion by josecastillagallardo** [ktbyers/netmiko#2982] [Q&A]
why ssh_autodetect.py sometimes detect device_type wrong and sometimes detect device_type fine for same device?

How can I that working always fine?

**Comment by ktbyers** [comment]
@josecastillagallardo You need to provide your code, what response you were expecting, and what response you received.

**Comment by josecastillagallardo** [comment]
def device_type(equipo):

	comando = ["C:\Program Files\PuTTY\plink.exe","-L","2220:"+equipo+":22","-P","22","-pw","pw","user@jumpserver"]
	ssh = subprocess.Popen(comando,shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
	#time.sleep(5)
	equipo_salto = {
		'device_type': 'autodetect',
		'host':   'localhost',
		'username': 'user',
		'password': 'pw',
		'port' : 2220,
		'global_delay_factor': 2,
	}
	try:
		guesser = SSHDetect(**equipo_salto)
		#guesser = SSHDetect(**equipo_salto)
		best_match = guesser.autodetect()
		#print(best_match)
		if not best_match:
			#best_match = 'terminal_server'
			print('Se vuelve a comprobar '+equipo)
			guesser = SSHDetect(**equipo_salto)	# TODO volvemos a hacerlo pasar para los casos de falsos positivos
			best_match = guesser.autodetect()
			if not best_match:
				best_match = 'generic_termserver'
		#Debug output of detection results
		print('Equipo: '+equipo+ ' device_type: ' + best_match)
		ssh.terminate()
		return(best_match)
	except (NetMikoTimeoutException):
		ssh.terminate()
		print ('\n'+'NetMikoTimeoutException: '+equipo)
	except (NetMikoAuthenticationException):
		ssh.terminate()
		print ('\n'+'NetMikoAuthenticationException: '+equipo)
	except (SSHException):
		ssh.terminate()
		print ('\n'+'SSHException: '+equipo)


For a juniper sometimes detect juniper_junos and other time None.
